File: 003.py
"""Find the largest prime factor of the number 600851475143"""
from math import floor, sqrt

NUM = 600851475143
INT_SQRT_NUM = int(floor(sqrt(NUM)))

def is_prime(n):
    """Determine if n is prime"""
    if n <= 3:
        if n <= 1:
            return False
        return True
    if n % 2 == 0 or n % 3 == 0:
        return False
    for j in xrange(5, int(floor(sqrt(n))) + 1, 6):
        if n % j == 0 or n % (j + 2) == 0:
            return False
    return True

for i in xrange(INT_SQRT_NUM, 1, -1):
    if NUM % i == 0 and is_prime(i):
        print i
        break

File: 003.swift
#!/usr/bin/xcrun swift

import Foundation

// Find the largest prime factor of the number 600851475143.

let number = 600_851_475_143
let sqrtOfNumber = Int(sqrt(Double(number)))

func isPrime(n: Int) -> Bool {
    // Determine if n is prime.
    assert(n >= 1, "N cannot be less than one")
    if n == 1 {
        return false
    } else if n == 2 || n == 3 {
        return true
    }
    for i in 2...Int(sqrt(Double(n))) {
        if n % i == 0 {
            return false
        }
    }
    return true
}

for var i = sqrtOfNumber; i > 1; --i {
    if number % i == 0 && isPrime(i) {
        println(i)
        break
    }
}
